CMakeLists.txt

```diff
@@ -535,6 +535,7 @@ list(APPEND CMAKE_PREFIX_PATH "${CMAKE_BINARY_DIR}")
 
 include(CTest)
 if(BUILD_TESTING)
+  set(CMAKE_CXX_STANDARD 20)
```

Comment:
@cngzhnp  - I think this will break the builds for users who are building library with tests using gcc4.8 ( or any C++11 only compiler) ?

Reply:
Yes, you are right. However, it is like CI/CD pipeline design problem for opentelemetry-cpp. There are a lot of scripts which setup different versions of third party libraries or submodules etc. All of them are not in one page.

From my perspective, if we still want to support GCC 4.8 version with C++11 standard, we need a better CMake/CI-CD/Script design for this. Otherwise, like me people might only focus on passing the CI tests but not the real usecases.

Reply:
This sets C++20 for all builds using tests, which breaks gcc4.8.

See related comment, to set C++14 is relevant ci/do_ci.sh targets instead.